app.go

```diff
@@ -135,7 +137,6 @@ func New(config Config) (*App, error) {
 			}
 
 			request.SetHeader("X-API-KEY", string(credentials.ApiKey))
-			zeroByteArray(credentials.ApiKey)
```

Comment:
Don't change the functionality.
I think it is the test code issue. 
https://github.com/cisco-pxgrid/cloud-sdk-go/blob/main/app_test.go#L69

The credential is created outside of the GetCredential func so it is being reused.
Put the credential inside the GetCredential func and give it a try.